flexmeasures/data/services/asset_grouping.py

```diff
@@ -0,0 +1,242 @@
+    if "renewables" in custom_additional_groups:
+        asset_queries["renewables"] = (
+            GenericAsset.query.join(GenericAssetType)
+            .filter(GenericAsset.generic_asset_type_id == GenericAssetType.id)
+            .filter(GenericAssetType.name.in_(["solar", "wind"]))
+        )
+    if "EVSE" in custom_additional_groups:
+        asset_queries["EVSE"] = (
+            GenericAsset.query.join(GenericAssetType)
+            .filter(GenericAsset.generic_asset_type_id == GenericAssetType.id)
+            .filter(GenericAssetType.name.in_(["one-way_evse", "two-way_evse"]))
+        )
+
+    # 2. We also include a group per asset type - using the pluralised asset type display name
+    for asset_type in GenericAssetType.query.all():
+        asset_queries[pluralize(asset_type.name)] = (
+            GenericAsset.query.join(GenericAssetType)
+            .filter(GenericAsset.generic_asset_type_id == GenericAssetType.id)
+            .filter(GenericAssetType.name == asset_type.name)
+        )
+
+    # 3. Finally, we group assets by location
+    if "location" in custom_additional_groups:
+        asset_queries.update(get_location_queries())
+
+    # only current user's account
+    asset_queries = limit_assets_to_account(asset_queries)
+
+    return asset_queries
+
+
+def get_location_queries() -> Dict[str, Query]:
+    """
+    We group EVSE assets by location (if they share a location, they belong to the same Charge Point)
+    Like get_asset_group_queries, the values in the returned dict still need an executive call, like all(), count() or first().
+
+    The Charge Points are named on the basis of the first EVSE in their list,
+    using either the whole EVSE display name or that part that comes before a " -" delimiter. For example:
+    If:
+        evse_display_name = "Seoul Hilton - charger 1"
+    Then:
+        charge_point_display_name = "Seoul Hilton (Charge Point)"
+
+    A Charge Point is a special case. If all assets on a location are of type EVSE,
+    we can call the location a "Charge Point".
+    """
+    asset_queries = {}
+    all_assets = GenericAsset.query.all()
+    loc_groups = group_assets_by_location(all_assets)
+    for loc_group in loc_groups:
+        if len(loc_group) == 1:
+            continue
+        location_type = "(Location)"
+        if all(
+            [
+                asset.asset_type.name in ["one-way_evse", "two-way_evse"]
+                for asset in loc_group
+            ]
+        ):
+            location_type = "(Charge Point)"
+        location_name = f"{loc_group[0].name.split(' -')[0]} {location_type}"
+        asset_queries[location_name] = GenericAsset.query.filter(
+            GenericAsset.name.in_([asset.name for asset in loc_group])
+        )
+    return asset_queries
+
+
+def limit_assets_to_account(
+    asset_queries: Union[Query, Dict[str, Query]]
+) -> Union[Query, Dict[str, Query]]:
+    """Filter out any assets that are not in the user's account."""
```

Comment:
I'm not sure what we'll use this code for in the future. Right now it's only used by the dashboard. Which shows by definition assets for your account. 

If anybody wants to use this later for other purposes, they could add a `limit_to_users_account` boolean parameter (which defaults to True). I'll add a TODO note for that possibility, but I'd like to keep doing that out of this PR as we are not sure where this code is going to be used. Maybe it will be used for admins to see other user's dashboards. Or for a new purpose where we want to group assets across accounts (not sure what a use case would be).

However, I got an idea from this discussion to improve this PR: I'll add the modern authorization check that you should be able to read asset info on your account.

Reply:
I just meant that, before project 9, on the dashboard admins could see assets from all accounts, whereas here it seems they can no longer see assets from other accounts.

Reply:
Mh, that might be a use case indeed.